files/sensu_slo.rb

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env ruby
+#
+# Sensu Handler: sensu_slo
+#
+# This handler calculate the age of a check by subtracting the execution
+# time of the check with the current time on the Sensu server. This can
+# be used to monitor check latency and also, by evaluating the interval
+# time, used to give and idea of the number of succesfull checks
+# navigating Sensu's message bus.
+
+require 'rubygems' if RUBY_VERSION < '1.9.0'
+require 'sensu-handler'
+require 'socket'
+
+$env_dir = "/nail/etc/"
+$default_dims_files = %w[ habitat ]
+
+class SensuSLOHandler < Sensu::Handler
+
+  def handle
+
+    metric_name = settings['sensu_slo']['metric_name'] || 'sensu.check_age'
+
+    statsite_host = settings['sensu_slo']['statsite_host'] || '127.0.0.1'
+    statsite_port = settings['sensu_slo']['statsite_port'] || 8125
+
+    # Get hostname of this Sensu server
+    begin
+        hostname = Socket.gethostname
+    rescue => _e
+        STDERR.print "Could not determine hostname: #{e}"
+        return
+    end
+
+    # Check required check fields are available
+    if @event["check"].nil?
+      STDERR.print "Check result does not appear to contain check data"
+      return
+    end
+
+    if @event["check"]["name"].nil?
+      STDERR.print "Check result did not have a 'name'"
+      return
+    end
+
+    if @event["client"]["name"].nil?
+      STDERR.print "Check result did not have a 'client name'"
+      return
+    end
+
+    # Extract the name of the check and the client which ran it
+    check_name = @event["check"]["name"]
+    client_name = @event["client"]["name"]
+
+    # Calculate how long ago the check was executed
+    now = Time.now.to_i
+    executed = @event["check"]["executed"].to_i
+    if executed == nil
+      STDERR.puts "Check result does not have an 'executed' field"
+      return
+    end
+    age = now - executed
+
+    # Create an array to hold the metric Dimensions
+    dims = Array.new
+    dims << ["metric_name", metric_name]
+
+    # Add default environment Dimensions
+    $default_dims_files.each do |f|
```

Comment:
we don't need a loop here as it's a 1 element array, right?

Reply:
It allow for easy extension if required. But sure, it is not technically required anymore